python/Dockerfile

```
@@ -81,13 +81,10 @@ RUN gem update --system && \
 ARG PYTHON_USER_BASE_BIN="$RUNNER_USER_HOME/.local/bin"
 ENV PATH $PYTHON_USER_BASE_BIN:$PATH
 
-# Install pip and Pipenv
-USER root
-COPY python/Pipfile /tmp/
-RUN cd /tmp && \
-    pip install pipenv && \
-    pipenv install --system --skip-lock --clear && \
-    rm /tmp/Pipfile && \
+# Install `pip` and `pipenv`
+COPY python/requirements.txt "${RUNNER_USER_HOME}/"
+RUN pip install "--requirement=${RUNNER_USER_HOME}/requirements.txt" --user --quiet --no-cache-dir && \
```

[nitpick] According to the [changelog](https://pip.pypa.io/en/stable/news/#id226), `--user` is no longer required since Pip 20.0.

```suggestion
RUN pip install "--requirement=${RUNNER_USER_HOME}/requirements.txt" --quiet --no-cache-dir && \
```